### PR TITLE
apollo explorer plugin

### PR DESCRIPTION
**Pull request by unredundant**
## Hey, I just made a Pull Request!

Hey 👋 This adds Apollo Explorer support to Backstage! This is just a PoC at the moment, but has an open feature request here #6692

Screen Shot 2022-07-12 at 10 56 07 PM

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/apollo-explorer/api-report.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+/// <reference types="react" />
+
+import { BackstagePlugin } from '@backstage/core-plugin-api';
+import { JSONObject } from '@apollo/explorer/src/helpers/types';
+import { RouteRef } from '@backstage/core-plugin-api';
+
+// Warning: (ae-missing-release-tag) "ApolloExplorerPage" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
```
Comment (benjdlambert) [inline plugins/apollo-explorer/api-report.md:12]: Think you need to add a `tsdoc` `@public` tag to the components that you're exporting.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Just some small things with the `api-report.md` and then I think it's looking pretty good.

**Comment by unredundant** [comment]
Awesome, I'll get that sorted some time today hopefully!  Question about the release process... by setting the following in the changeset tool

```
---
'@backstage/plugin-apollo-explorer': minor
---

Apollo Explorer plugin now available! Installation instructions can be found in the plugin README
```

will the version automatically be bumped from `0.0.0 ~> 0.1.0` on the next backstage release? 

Just want to make sure we don't roll out `0.0.0` which it is currently set to

**Comment by benjdlambert** [comment]
Yep that's indeed the idea! :pray: Next release is tomorrow. They happen every tuesday, so if you can get it in by tomorrow morning there's a good chance that we can get it shipped quickly.

File: plugins/apollo-explorer/package.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  "dependencies": {
+    "@backstage/core-components": "^0.10.0-next.3",
+    "@backstage/core-plugin-api": "^1.0.4-next.0",
+    "@backstage/theme": "^0.2.16-next.1",
+    "@material-ui/core": "^4.9.13",
+    "@material-ui/icons": "^4.9.1",
+    "@material-ui/lab": "^4.0.0-alpha.57",
+    "react-use": "^17.2.4",
+    "@apollo/explorer": "^1.1.1",
+    "use-deep-compare-effect": "^1.8.1"
```
Comment (benjdlambert) [inline plugins/apollo-explorer/package.json:33]: Don't think we use this dependency do we? Can we remove it?

**Review by benjdlambert — COMMENTED** [review]
Spotted one last thing

File: plugins/apollo-explorer/package.json (hunk 2 above)
Comment (unredundant) [reply plugins/apollo-explorer/package.json:33]: Ya I only added this b/c I got an error that it was missing 🤷 let me quickly try running again w/out it

Comment (unredundant) [reply plugins/apollo-explorer/package.json:33]: Screen Shot 2022-07-18 at 10 04 18 AM

Comment (benjdlambert) [reply plugins/apollo-explorer/package.json:33]: Well that is annoying. OK. No worries, thanks for checking!

**Review by benjdlambert — APPROVED** [review]
Thanks for this! :pray:

**Comment by unredundant** [comment]
Thanks for the speedy reviews 🚀

**Pull request merged**